cli/organizationsettings.go

```diff
@@ -0,0 +1,159 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) organizationSettings(orgContext *OrganizationContext) *serpent.Command {
+	cmd := &serpent.Command{
+		Use:     "settings",
+		Short:   "Manage organization settings.",
+		Aliases: []string{"setting"},
+		Handler: func(inv *serpent.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+		Hidden: true,
+		Children: []*serpent.Command{
+			r.printOrganizationSetting(orgContext),
+			r.updateOrganizationSetting(orgContext),
+		},
+	}
+	return cmd
+}
+
+func (r *RootCmd) updateOrganizationSetting(orgContext *OrganizationContext) *serpent.Command {
+	client := new(codersdk.Client)
+	cmd := &serpent.Command{
+		Use:   "set <groupsync | rolesync>",
+		Short: "Update specified organization setting.",
+		Long: FormatExamples(
+			Example{
+				Description: "Update group sync settings.",
+				Command:     "coder organization settings set groupsync < input.json",
+			},
+		),
+		Options: []serpent.Option{},
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(1),
+			r.InitClient(client),
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+			org, err := orgContext.Selected(inv, client)
+			if err != nil {
+				return err
+			}
+
+			// Read in the json
+			inputData, err := io.ReadAll(inv.Stdin)
+			if err != nil {
+				return xerrors.Errorf("reading stdin: %w", err)
+			}
+
+			var setting any
+			switch strings.ToLower(inv.Args[0]) {
```

Comment:
This feels like we are reinventing cobra sub commands. Could we compose these as true sub commands?

Reply:
Fixed:
```
USAGE:
  coder organizations settings set

  Update specified organization setting.

    - Update group sync settings.:
  
        $ coder organization settings set groupsync < input.json 

SUBCOMMANDS:
    group-sync    Group sync settings to sync groups from an IdP.
    role-sync     Role sync settings to sync organization roles from an IdP.

———
Run `coder --help` for a list of global options.
```